src/cpp/rtps/xmlparser/XMLDynamicParser.cpp

```diff
@@ -724,6 +724,12 @@ XMLP_ret XMLParser::parseXMLStructDynamicType(
      */
     XMLP_ret ret = XMLP_ret::XML_OK;
     const char* name = p_root->Attribute(NAME);
+    if (!name)
```

Comment:
NIT, compare explicitly against nullptr for uniformity
```suggestion
    if (nullptr == name)
```